src/js/angular/resources/chart-models/chart-data.js

```diff
@@ -51,25 +55,16 @@ export class ChartData {
         });
     }
 
-    /**
-     * Must implement
-     * @return {string} the title name
-     */
-    getTitle() {
-        return 'No chart title set';
-    }
-
-    setTitle(title) {
+    setSubTitle(keyValues) {
         this.chartOptions.title.enable = true;
-        this.chartOptions.title.text = title;
-    }
-
-    getSubTitle() {
-    }
-
-    setSubTitle(subTitle) {
-        this.chartOptions.subtitle.enable = true;
-        this.chartOptions.subtitle.html = subTitle;
+        const subTitle = keyValues.map((keyValue) => {
```

Comment:
If subTitle is actually an array, you can rename it to plural or something

Reply:
renamed